### PR TITLE
feat: optional close button on last pagination page on compact pagination

### PR DESCRIPTION
**Pull request by kemuru**
visual demonstration:

https://github.com/kleros/ui-components-library/assets/102478601/c1bf88c2-825c-4448-8a5d-20b707c1b373

---

## PR-Codex overview
The focus of this PR is to update the pagination functionality by adding a new optional callback for closing on the last page.

### Detailed summary
- Updated `use-pagination` hook to accept an optional `onCloseOnLastPage` callback.
- Added a new `CloseButton` component with an `onClick` event to trigger the `onCloseOnLastPage` callback.
- Updated `CompactPagination` component to conditionally render the `CloseButton` based on the current page being the last page.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

## Summary by CodeRabbit

- **New Features**
  - Added an optional `onCloseOnLastPage` function prop to `CompactPagination`, allowing users to specify an action when the last page is reached.
  - Enhanced `usePagination` hook to accept an `onCloseOnLastPage` parameter for customized behavior on the last page.

### CODE REVIEW

File: src/hooks/pagination/use-pagination.tsx [hunk 1]
```diff
@@ -3,11 +3,16 @@ const usePagination = (
   numPages: number,
   //eslint-disable-next-line @typescript-eslint/ban-types
   callback: Function,
+  closeOnLastPage?: () => void,
```
[the remark this answers is not in the document]
Comment (kemuru) [reply src/hooks/pagination/use-pagination.tsx:6]: unrelated to this pr

File: src/hooks/pagination/use-pagination.tsx [hunk 2]
```diff
@@ -3,11 +3,16 @@
...
   numNeighbors = 2
 ) => {
   const incrementPage = () => {
-    const newPage = Math.min(numPages, currentPage + 1);
-    callback(newPage);
+    if (currentPage === numPages && closeOnLastPage) {
+      closeOnLastPage();
```
[the remark this answers is not in the document]
Comment (kemuru) [reply src/hooks/pagination/use-pagination.tsx:11]: i mean thats already checked in the "if"

File: src/lib/pagination/compact.tsx [hunk 3]
```diff
@@ -70,21 +86,28 @@ const CompactPagination: React.FC<CompactPaginationProps> = ({
...
       </LeftArrow>
-      <RightArrow disabled={maxPageReached} onClick={incrementPage}>
-        <Arrow className={StyledSVG.styledComponentId} />
-      </RightArrow>
+      {currentPage === numPages && onCloseOnLastPage ? (
+        <CloseButton onClick={onCloseOnLastPage}>
+          <SolidErrorIcon className={StyledSVG.styledComponentId} />
+        </CloseButton>
+      ) : (
+        <RightArrow disabled={maxPageReached} onClick={incrementPage}>
```
Comment (Harman-singh-waraich) [inline src/lib/pagination/compact.tsx:107]: is `disabled` prop needed now, if not we can remove this logic

File: src/lib/pagination/compact.tsx [hunk 4]
```diff
@@ -57,11 +58,26 @@ const RightArrow = styled(ArrowButton)`
...
+      fill: ${({ theme }) => theme.klerosUIComponentsSecondaryBlue};
+    }
+  }
+`;
+
 interface CompactPaginationProps {
   currentPage: number;
   numPages: number;
   //eslint-disable-next-line @typescript-eslint/ban-types
   callback: Function;
```
Comment (alcercu) [inline src/lib/pagination/compact.tsx:79]:
```suggestion
  callback: (newPage: number) => void;
```

File: src/lib/pagination/compact.tsx [hunk 5]
```diff
@@ -70,21 +86,28 @@
   currentPage,
   numPages,
   callback,
+  onCloseOnLastPage,
```
[the remark this answers is not in the document]
Comment (alcercu) [reply src/lib/pagination/compact.tsx:89]: It's an optional param

File: src/lib/pagination/compact.tsx (hunk 4 above)
Comment (alcercu) [inline src/lib/pagination/compact.tsx:79]:
```suggestion
  callback: (newPage: number) => void;
```

**Review by alcercu — APPROVED** [review]
lgtm

**Pull request merged**